src/utils.js

```diff
@@ -32,18 +39,18 @@ export function getDisplayName(WrappedComponent: string | Component): string {
 }
 
 export function validateSpacingProps(
-  props: GridProps | LayoutProps,
+  props: SpacingProps,
   type: 'margin' | 'padding'
 ) {
   if (process.env.NODE_ENV !== 'production') {
-    const values = pick(props, directions.map(dir => type + capitalize(dir)))
-
-    // flow complains about props since `padding` cannot be found on `LayoutProps`
-    const value = (props: any)[type]
+    const values = getSpacingValues(props, type)
+    // flow is having trouble understanding what `props[type]` could be
+    const value: any = props[type]
```

Comment:
can we cast it? something like `const (value: any): SpacingValues = props[type]`. That way we don't need the comment and flow knows what it's getting